allennlp/data/dataset_readers/coreference_resolution/conll.py

```diff
@@ -0,0 +1,192 @@
+    def text_to_instance(self,  # type: ignore
+                         doc_key: str,
+                         sentences: List[List[str]],
+                         clusters: List[List[Tuple[int, int]]]) -> Instance:
+        # pylint: disable=arguments-differ
+        flattened_sentences = [t for s in sentences for t in s]
+        text_field = TextField([Token(t) for t in flattened_sentences], self._token_indexers)
+
+        cluster_dict = {}
+        for cluster_id, cluster in enumerate(clusters):
+            assert len(cluster) > 1
+            for mention in cluster:
+                cluster_dict[tuple(mention)] = cluster_id
+
+        span_labels = []
+        span_starts = []
+        span_ends = []
+        sentence_offset = 0
+        for sentence in sentences:
+            for i in range(len(sentence)):
+                for j in range(i, min(i + self._max_span_width, len(sentence))):
+                    start = sentence_offset + i
+                    end = sentence_offset + j
+                    if (start, end) in cluster_dict:
+                        span_labels.append(cluster_dict[(start, end)])
+                    else:
+                        span_labels.append(-1)
+                    start_field: Field = IndexField(start, text_field)
+                    end_field: Field = IndexField(end, text_field)
+                    span_starts.append(start_field)
+                    span_ends.append(end_field)
+            sentence_offset += len(sentence)
+
+        span_starts_field = ListField(span_starts)
+        span_ends_field = ListField(span_ends)
+        span_labels_field = SequenceLabelField(span_labels, span_starts_field)
+
+        metadata_field = MetadataField({"doc_key" : doc_key,
+                                        "clusters": clusters,
+                                        "text": flattened_sentences})
+
+        fields: Dict[str, Field] = {'text': text_field,
+                                    'span_starts': span_starts_field,
+                                    'span_ends': span_ends_field,
+                                    'span_labels': span_labels_field,
+                                    'metadata': metadata_field}
+        return Instance(fields)
+
+    @classmethod
+    def from_params(cls, params: Params) -> 'ConllCorefReader':
+        token_indexers = TokenIndexer.dict_from_params(params.pop('token_indexers', {}))
+        max_span_width = params.pop('max_span_width')
+        params.assert_empty(cls.__name__)
+        return cls(token_indexers=token_indexers, max_span_width=max_span_width)
+
+    @staticmethod
+    def get_doc_key(doc_id, part):
+        return "{}_{}".format(doc_id, int(part))
+
+    @staticmethod
+    def normalize_word(word):
+        if word == "/." or word == "/?":
+            return word[1:]
+        else:
+            return word
+
+    def handle_line(self, line, document_state):
+        begin_document_match = self._begin_doc_regex.match(line)
```

Comment:
Yep, this can be avoided. Though this isn't the slow preprocessing that I was referring to. It's the stuff that's in the scripts directory that reconciles ontonotes with the conll data. This code is reasonably quick to run (less than a minute).